### DISCUSSION TITLE
Dynamic execution nodes - Pbspro

### DISCUSSION THREAD

**Discussion by sidharthjs** [GoogleCloudPlatform/cluster-toolkit#1544] [Q&A] [answered]
I have been setting up a pbspro cluster using the following modules:

* pbspro-client
* pbspro-server
* pbspro-execution

The `pbspro-execution` has `instance_count` input to specify the count of the nodes but does not seem to have max dynamic node count input like its available in slurm-gcp-v5-node/partition. I would like to have a setup where the execution nodes get autoscaled dynamically based on the number of workloads in a given time. Is it possible with the currently available modules? Or is the feature planned for later? Thanks.

Any help would be appreciated.

**Comment by tpdownes** [comment, marked as answer]
Thanks for submitting this feature request. I'm glad you were able to make the modules work as they are not yet prominently documented. This is functionality we are considering adding to the module, but we do not have an anticipate date.

I might suggest the following as a stopgap:

1. Provision with N>0 execution hosts
2. Run your workload.
3. Modify the blueprint to have N=0 execution hosts
4. Run `ghpc create -w ...`
5. Run `ghpc deploy ...`

I have not tested this, but I believe it may scale down your execution hosts without destroying the rest of your infrastructure. If you try this, please update the discussion.

**Reply by sidharthjs** [reply]
Alright. Thanks @tpdownes. 
Let me update if the scale-down works when I do it.
